code/modules/clothingbooth/clothingbooth.dm

```diff
@@ -0,0 +1,446 @@
+	cash_to_add.amount = 0
+	user.visible_message(SPAN_NOTICE("[user.name] inserts credits into [src]"))
+	user.u_equip(cash_to_add)
+	cash_to_add.dropped(user)
+	playsound(user, 'sound/machines/capsulebuy.ogg', 80, TRUE)
+	qdel(cash_to_add)
+
+/obj/machinery/clothingbooth/proc/eject_cash(turf/location, mob/target)
+	if (src.cash <= 0)
+		return
+	if (!location)
+		location = get_turf(src)
+	if (ismob(target))
+		target.put_in_hand_or_drop(new /obj/item/currency/spacecash(location, src.cash))
+	else
+		new /obj/item/currency/spacecash(location, src.cash)
+	src.cash = 0
+
+/// A shame this will rarely fire, since having insufficient funds will just disable the purchase button.
+/obj/machinery/clothingbooth/proc/insufficient_funds()
+	boutput(usr, SPAN_ALERT("Insufficient funds!"))
+	animate_shake(src, 12, 3, 3)
+	playsound(src, 'sound/impact_sounds/Metal_Hit_Heavy_1.ogg', 50, TRUE)
+
+/obj/machinery/clothingbooth/proc/purchase_item(mob/target)
+	var/purchased_item_path = src.selected_item.item_path
+	if (!purchased_item_path)
+		return
+	var/obj/item/clothing/purchased_item = new purchased_item_path(src)
+	if (src.dye)
+		purchased_item.color = src.dye
+	target.put_in_hand_or_drop(purchased_item)
+
+/obj/machinery/clothingbooth/proc/add_occupant(mob/target)
+	if (src.occupant)
+		return
+	target.set_loc(src.loc)
+	SPAWN(0.5 SECONDS)
+		if (src.occupant)
+			return
+		src.icon_state = "clothingbooth-close"
+		target.set_loc(src)
+		src.occupant = target
+		src.preview.add_client(target.client)
+		src.equip_and_preview()
+		src.ui_interact(target)
+
+/obj/machinery/clothingbooth/proc/remove_occupant(mob/target)
+	if (!src.occupant)
+		return
+	src.icon_state = "clothingbooth-open"
+	SPAWN(2 SECONDS)
+		src.eject_contents(target)
+
+/obj/machinery/clothingbooth/proc/eject_contents(mob/target)
+	src.clear_preview_item()
+	src.preview.remove_all_clients()
+	tgui_process.close_uis(src)
+	src.reset_clothingbooth_parameters()
+	var/turf/T = get_turf(src)
+	if (!target && src.occupant)
+		target = src.occupant
+	if (target?.loc == src) // Ensure mob wasn't otherwise removed during out spawn call.
+		target.set_loc(T)
+		src.eject_cash(T, target)
+		for (var/obj/item/I in src.contents)
+			target.put_in_hand_or_drop(I)
+		for (var/atom/movable/AM in contents)
+			AM.set_loc(T)
+	else
+		src.eject_cash(T)
+		for (var/atom/movable/AM in contents)
+			AM.set_loc(T)
+	src.occupant = null
+
+// Blatantly stolen from `/datum/component/barber`.
+/obj/machinery/clothingbooth/proc/reference_clothes(mob/living/carbon/human/to_copy, mob/living/carbon/human/to_paste)
+	src.nullify_clothes(to_paste)
+	to_paste.wear_suit = SEMI_DEEP_COPY(to_copy.wear_suit)
+	to_paste.w_uniform = SEMI_DEEP_COPY(to_copy.w_uniform)
+	to_paste.shoes = SEMI_DEEP_COPY(to_copy.shoes)
+	to_paste.gloves = SEMI_DEEP_COPY(to_copy.gloves)
+	to_paste.glasses = SEMI_DEEP_COPY(to_copy.glasses)
+	to_paste.head = SEMI_DEEP_COPY(to_copy.head)
+	to_paste.wear_id = SEMI_DEEP_COPY(to_copy.wear_id)
```

Comment:
`SEMI_DEEP_COPY()` is an *extremely* unsafe proc, and this has been the root cause of thousands of runtimes in each round since this got testmerged, primarily caused by issues with PDAs in the wear_id slot getting `SEMI_DEEP_COPY()`ed